src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudWindowsIT.java

```diff
@@ -200,37 +218,70 @@ public void testWorkerCreated() throws Exception {
         assertEquals(logs(),3, i.getLabels().size());
 
         // Instance should have a label with key CONFIG_LABEL_KEY and value equal to the config's name prefix
-        assertEquals(logs(), validInstanceConfiguration1().namePrefix, i.getLabels().get(ComputeEngineCloud.CONFIG_LABEL_KEY));
+        assertEquals(logs(), ic.namePrefix, i.getLabels().get(ComputeEngineCloud.CONFIG_LABEL_KEY));
         // proper id label to properly count instances
         assertEquals(logs(), cloud.getInstanceUniqueId(), i.getLabels().get(ComputeEngineCloud.CLOUD_ID_LABEL_KEY));
     }
 
-    private static InstanceConfiguration validInstanceConfiguration1() throws Exception{
-        GoogleKeyPair kp = GoogleKeyPair.generate("");
-        // Have to reformat since GoogleKeyPair's format is for metadata server and not typical public key format
-        String publicKey = kp.getPublicKey().trim().substring(1);
+    // Tests snapshot is created when we have failure builds for given node
+    // Snapshot creation is longer for windows vm's.
+    @Test(timeout = 0)
+    public void testSnapshotCreated() throws Exception {
```

Comment:
Reading through this, it seems like this is only testing Snapshot creation on Windows VMs? (Please correct if I'm wrong). To make sure we don't get caught by weird OS based edge cases, I suggest having two separate test cases, one for Linux and Windows each.

Reply:
ComputeEngineCloudIT is for Linux

Reply:
Ah ok, I see what you mean. Thanks for clarifying.

Looking at the two classes, it looks like there's a lot of duplication which could be resolved through some parameterization, this leads to more maintenance work as whenever we want to update an int-test we have to update both sets.. Ideally we should have a single set of int-tests which different configs get passed through. Could you please do me a favor and a TODO linking to this issue in both IT classes: https://issues.jenkins-ci.org/browse/JENKINS-56163

Reply:
Please make sure to accompany conversation resolutions either with the corresponding code modifications, or else a response in the conversation. Thanks.

Reply:
To-do added. Done.

Reply:
Speaking of which, I also would like to re-use instances in the IT's because currently the tests take over 10 minutes to run sometimes just because we keep provisioning new nodes even though some tests re-use the same configurations.